plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.test.ts

```diff
@@ -59,94 +55,169 @@ const testFactRetriever: FactRetriever = {
     ];
   }),
 };
-const cadence = '1 * * * *';
+const defaultCadence = '1 * * * *';
 describe('FactRetrieverEngine', () => {
   let engine: FactRetrieverEngine;
-  let factSchemaAssertionCallback: (
+  type FactSchemaAssertionCallback = (
     factSchemaDefinition: FactSchemaDefinition,
   ) => void;
-  let factInsertionAssertionCallback: (facts: TechInsightFact[]) => void;
 
-  const mockRepository: TechInsightsStore = {
-    insertFacts: (facts: TechInsightFact[]) => {
-      factInsertionAssertionCallback(facts);
-      return Promise.resolve();
-    },
-    insertFactSchema: (def: FactSchemaDefinition) => {
-      factSchemaAssertionCallback(def);
-      return Promise.resolve();
-    },
-  } as unknown as TechInsightsStore;
+  jest.setTimeout(15000);
 
-  const mockFactRetrieverRegistry: FactRetrieverRegistry = {
-    listRetrievers(): FactRetriever[] {
-      return [testFactRetriever];
-    },
-    listRegistrations(): FactRetrieverRegistration[] {
-      return [{ factRetriever: testFactRetriever, cadence }];
-    },
-  } as unknown as FactRetrieverRegistry;
+  type FactInsertionAssertionCallback = ({
+    facts,
+    id,
+  }: {
+    id: string;
+    facts: TechInsightFact[];
+  }) => void;
 
-  const defaultEngineConfig = {
-    factRetrieverContext: {
-      logger: getVoidLogger(),
-      config: ConfigReader.fromConfigs([]),
-      discovery: {
-        getBaseUrl: (_: string) => Promise.resolve('http://mock.url'),
-        getExternalBaseUrl: (_: string) => Promise.resolve('http://mock.url'),
+  function createMockRepository(
+    insertCallback: FactInsertionAssertionCallback,
+    assertionCallback: FactSchemaAssertionCallback,
+  ): TechInsightsStore {
+    return {
+      insertFacts: ({
+        facts,
+        id,
+      }: {
+        facts: TechInsightFact[];
+        id: string;
+      }) => {
+        insertCallback({ facts, id });
+        return Promise.resolve();
       },
-    },
-    factRetrieverRegistry: mockFactRetrieverRegistry,
-    repository: mockRepository,
-  };
+      insertFactSchema: (def: FactSchemaDefinition) => {
+        assertionCallback(def);
+        return Promise.resolve();
+      },
+    } as unknown as TechInsightsStore;
+  }
 
-  it('Should update fact retriever schemas on initialization', async () => {
-    factSchemaAssertionCallback = ({ id, schema, version, entityFilter }) => {
-      expect(id).toEqual('test-factretriever');
-      expect(version).toEqual('0.0.1');
-      expect(entityFilter).toEqual([{ kind: 'component' }]);
-      expect(schema).toEqual({
-        testnumberfact: {
-          type: 'integer',
-          description: '',
-        },
-      });
-    };
+  function createMockFactRetrieverRegistry(
+    cadence?: string,
+    factRetriever?: FactRetriever,
+  ): FactRetrieverRegistry {
+    const cron = cadence ? cadence : defaultCadence;
```

Comment:
fancy syntax hint: `const cron = cadence ?? defaultCadence` (same as `||`, but only matches on null and undefined)

Reply:
Coming from a primarily Golang background here, all the syntactic sugar has my head spinning haha